.github/workflows/benchmarks.yml

```diff
@@ -150,15 +183,50 @@ jobs:
 
       - name: Upload benchmark result
         uses: actions/upload-artifact@v4
-        if: always()
         with:
           name: benchmark-result
           path: ${{ env.WORKING_DIRECTORY }}/${{ env.BENCHMARK_RESULT }}
           if-no-files-found: error
 
+      # The next section injects CPU profile collected by apmbench into the build.
+      # By copying the profile, uploading it to the artifacts and pushing it
+      # via a PR to update default.pgo.
+
+      - name: Copy CPU profile
+        run: make cp-cpuprof
+      
+      - name: Upload CPU profile
+        uses: actions/upload-artifact@v4
+        with:
+          name: cpu-profile
+          path: ${{ env.WORKING_DIRECTORY }}/${{ env.BENCHMARK_CPU_OUT }}
+          if-no-files-found: error
+
+      - name: Open PGO PR
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: |
+          if [ "$RUN_STANDALONE" = true ]; then
```

Comment:
ditto.

OTOH, do you think this snippet could be added to the Makefile?

Reply:
Good point, will move this to Makefile

Reply:
updated